public/components/notebooks/components/paragraph_components/para_input.tsx

```diff
@@ -49,7 +49,8 @@ export const ParaInput = (props: {
   para: ParaType;
   index: number;
   runParaError: boolean;
-  textValueEditor: (evt: React.ChangeEvent<HTMLTextAreaElement>, index: number) => void;
+  text: string;
+  onTextChange: (evt: React.ChangeEvent<HTMLTextAreaElement>, index: number) => void;
```

Comment:
<ParaInput> converted into "model" object, with input (`text`) and onChange (`onTextChange`) - `textValueEditor` was causing Paragraph re-render on change... this method eliminates the over-reach state change for text input.